File: Sort/BubbleSort/BubbleSort.c
//
//  BubbleSort.c
//  BubbleSort
//
//  Created by saidicaprio on 2018/12/26.
//  Copyright © 2018 saidicaprio. All rights reserved.
//

#include "BubbleSort.h"

void swap(int *a, int *b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

void BubbleSort(int arr[], int len) {
    for (int i = 0; i < len - 1; i++) {
        for (int j = 0; j < len - 1 - i; j++) {
            if (arr[j] > arr[j+1]) {
                swap(&arr[j], &arr[j+1]);
            }
        }
    }
}

File: LeetCode/Sort/BubbleSort.swift
//
//  BubbleSort.swift
//  Sort
//
//  Created by saidicaprio on 2019/5/21.
//  Copyright © 2019 saidicaprio. All rights reserved.
//


func bubbleSort<T: Comparable>(_ array: inout [T]) {
    guard array.count > 1 else { return }
    
    let n = array.count
    for i in 0 ..< n {
        for j in 0 ..< n - 1 - i {
            if array[j] > array[j+1] {
                (array[j], array[j+1]) = (array[j+1], array[j])
            }
        }
    }
}
